src/hyperloglog.c

```diff
@@ -1258,7 +1258,7 @@ void pfcountCommand(client *c) {
      *
      * The user specified a single key. Either return the cached value
      * or compute one and update the cache. */
-    o = lookupKeyWrite(c->db,c->argv[1]);
+    o = lookupKeyRead(c->db,c->argv[1]);
```

Comment:
let's start a separate discussion on this one, and the implications of doing lookupKeyRead.
First, the main difference here is how it behaves when used in a replica on an expired key, right?
in the past it would have meant that a user could access an expired key successfully.

now after our change, that difference between lookupKeyRead and lookupKeyWrite is gone (both do expiration check), and the main difference is stats and notifications.

The above is true for all the other places we changed between these functions (e.g. SUNIONSTORE), but what's special about this command is that it's a READONLY command, but it **does** modify the value (and can cause some inconsistency between the master and the replica).

But that fact mentioned above is not changed by our PR.
So the question is if anyone sees any other special concern about this command in this PR?

Reply:
I looked up how PFCOUNT works. It is using and updating a cache which is stored in the key, which is also accessible as a string using GET and SET. I can see two alternatives:

1. If the cache is invalid, compute the pfcount from the HLL but don't update the cache if it's read-only replica (or if writes are forbidden for any other reason, like what?). This might make the HLL a bit slow until the master has propagated an updated pfcount cache to the replica.
2. Remove the readonly flag from PFCOUNT and mark it as a write command instead.

Is (1) acceptable?

Reply:
OK, sounds good. We'll stick with (3) and I'll see if I can come up with a test case for the bugfix. :-)